core/src/test/java/com/ritense/valtimo/web/rest/ProcessResourceIntTest.java

```diff
@@ -0,0 +1,78 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.web.rest;
+
+import com.ritense.valtimo.BaseIntegrationTest;
+import org.camunda.bpm.engine.RepositoryService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import javax.inject.Inject;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class ProcessResourceIntTest extends BaseIntegrationTest {
+
+    @Inject
+    private ProcessResource processResource;
+
+    @Inject
+    public RepositoryService repositoryService;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void init() {
+        mockMvc = MockMvcBuilders.standaloneSetup(processResource).build();
+    }
+
+    @Test
+    void shouldGetProcessDefinitions() throws Exception {
+        mockMvc.perform(get("/api/process/definition")
+                .accept(APPLICATION_JSON_VALUE))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.[0].key").value("one-task-process"))
+            .andExpect(jsonPath("$.[0].readOnly").value(false))
+            .andExpect(jsonPath("$.[1].key").value("test-process"))
+            .andExpect(jsonPath("$.[1].readOnly").value(true));
+    }
+
+    @Test
+    void shouldGetProcessDefinitionXml() throws Exception {
+        var processDefinition = repositoryService.createProcessDefinitionQuery()
+            .processDefinitionKey("test-process")
+            .latestVersion()
+            .singleResult();
+
+        mockMvc.perform(get("/api/process/definition/{processDefinitionId}/xml", processDefinition.getId())
+                .accept(APPLICATION_JSON_VALUE))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.id").value(processDefinition.getId()))
+            .andExpect(jsonPath("$.bpmn20Xml").isNotEmpty())
+            .andExpect(jsonPath("$.readOnly").value(true))
+            .andExpect(jsonPath("$.systemProcess").value(true));
+    }
```

Comment:
I would expect a test to see if an update to a read-only processdefinition throws an error.

Reply:
I added a test. But not for your use-case because I found an issue: 
The FE calls this endpoint for deploying a process:  `/camunda-rest/engine/default/deployment/create`. This is a camunda endpoint and overriding it is a huge pain. Another solution is to make a new endpoint in Valtimo, which would be a breaking change. 

I didn't do either. So for now its only FE that makes sure that people dont deploy a readOnly process. and leave this problem for later :P

Reply:
I'll create a story to cover this from a BE perspective as well in major 10.